### DISCUSSION TITLE
[Feature Request] Export some log or metric about uplink and downlink volume usage per target host

### DISCUSSION THREAD

**Discussion by forsaken628** [v2fly/v2ray-core#2309] [Ideas]
So I can optimize my route policy to reduce unnecessary proxy channel usage.

**Comment by mydogshitgold** [comment]
I think `iftop` can do what you want.
```
sudo iftop -i lo
```

**Reply by forsaken628** [reply]
I prefer domain names, because it is impossible to know how many IPs belong to a CDN.

**Reply by mydogshitgold** [reply]
Or you can use `wireshare`.
It can use captured DNS packet data for name resolution
```bash
wireshark -N mdn
```

**Reply by forsaken628** [reply]
Wait, all traffic will be sent to the proxy host not target host, it's no work.

**Reply by mydogshitgold** [reply]
You need to monitor loopback interface to see your target hosts.
It would be easier if you are using transparent proxy.

**Reply by forsaken628** [reply]
Directly dialing the target host never happens on the client side, and the proxy host is not under my control.  Yes, transparent proxy is not used.

**Reply by mydogshitgold** [reply]
> Directly dialing the target host never happens on the client side

You can still capture the traffic to target hosts on **loopback** (also called **lo**) or **tun** interfaces, connections are made from there to your local v2ray node.
```
 browser <-> loopback/tun <-> client v2ray <-> server v2ray <-> target host
```

**Reply by forsaken628** [reply]
OK, but compared with logging at the application level, it is too complicated. Is there no plan to improve observability?
